src/trg-stats-dialog.c

```diff
@@ -95,6 +96,9 @@ static void trg_stats_dialog_set_property(GObject *object, guint property_id, co
 
 static void trg_stats_response_cb(GtkDialog *dlg, gint res_id, gpointer data G_GNUC_UNUSED)
 {
+    TrgStatsDialog *trg_dlg = TRG_STATS_DIALOG(dlg);
+    if (trg_dlg->update_stats_timer_tag)
+        g_source_remove(trg_dlg->update_stats_timer_tag);
```

Comment:
You need to also set it back to `0`.

You can use `g_clear_handle_id(&trg_dlg->update_stats_timer_tag, g_source_remove);` or do it manually.

(I forget which glib version this depends on)

Reply:
done, we were already using g_clear_handle_id() in other places, so I'm not worrying about glib version.